src/lib/network_pool/indexed_pool.ml

```diff
@@ -1170,16 +1287,16 @@ let add_from_backtrack :
           Map_set.insert
             (module Transaction_hash.User_command_with_valid_signature)
             t.all_by_fee fee_per_wu cmd
-      ; all_by_hash =
-          Map.set t.all_by_hash
-            ~key:(Transaction_hash.User_command_with_valid_signature.hash cmd)
-            ~data:cmd
+      ; all_by_hash = Map.set t.all_by_hash ~key:cmd_hash ~data:cmd
       ; applicable_by_fee =
           Map_set.insert
             (module Transaction_hash.User_command_with_valid_signature)
             t.applicable_by_fee fee_per_wu cmd
       ; transactions_with_expiration =
           add_to_expiration t.transactions_with_expiration cmd
+      ; entry_times = Map.add_exn t.entry_times ~key:cmd_hash ~data:entry_time
+      ; all_by_entry_time =
+          Map.add_exn t.all_by_entry_time ~key:entry_time ~data:cmd_hash
```

Comment:
Do we need to worry about an exception here if (unlikely) we add 2 transactions within the same nanosecond?

Reply:
I could change this to a set, but it seems unlikely that we'd be able to add 2 transactions within a nanosecond of one another.